addon/components/container-query.js

```diff
@@ -20,6 +20,13 @@ export default class ContainerQueryComponent extends Component {
     return this.args.debounce ?? 0;
   }
 
+  constructor() {
+    super(...arguments);
+
+    // The dynamic tag is restricted to be immutable
+    this.tagName = this.args.tagName || 'div';
```

Comment:
I used the OR operator `||` so that, in addition to `null` and `undefined` (which can be covered by the nullish coalescing operator `??`), the empty string `''` would also result in the default `<div>` tag.

Reply:
Your note on setting this in the constructor and not letting it be dynamic I think is pretty sensible 👍